packages/rule-tester/src/types/DependencyConstraint.ts

```diff
@@ -6,12 +6,12 @@ export interface RangeOptions {
 }
 
 export interface SemverVersionConstraint {
-  readonly range: string;
   readonly options?: RangeOptions | boolean;
+  readonly range: string;
 }
 export type AtLeastVersionConstraint =
-  | `${number}.${number}.${number}-${string}`
   | `${number}.${number}.${number}`
+  | `${number}.${number}.${number}-${string}`
```

Comment:
weird that this sorted this way.

Why does `-` come AFTER "nothing", but `.` comes BEFORE nothing.
Is this a bug? What the what?

Reply:
I think it's that <code>`</code> is being treated as before `-` alphabetically?

Reply:
That must be it - it must be considering the quote character as part of the thing.

Would be worth filing a bug for that.
"if the two types are the same literal type (eg both template literal types) then the quote character should be excluded from the sort comparison.

Reply:
👍 noted in https://github.com/typescript-eslint/typescript-eslint/issues/8479#issuecomment-2307391859